master/networking/ip-autodetection.md

```diff
@@ -0,0 +1,199 @@
+  ```
+
+Where auto-detection methods are based on:
+
+- **IP or domain name**
+
+  A reachable destination (IP address or domain). For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "can-reach=8.8.8.8"}]}]}}}}'
+
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "can-reach=www.google.com"}]}]}}}}'
+  ```
+
+- **Including matching interfaces**
+
+  A regular expression in golang syntax that includes interfaces that match. For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "interface=eth.*"}]}]}}}}'
+  ```
+
+- **Excluding matching interfaces**
+
+  A regular expression in golang syntax that excludes interfaces that match. For example:
+
+  ```
+  kubectl patch ds -n kube-system calico-node --patch \
+      '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP_AUTODETECTION_METHOD", "value": "skip-interface=eth.*"}]}]}}}}'
+  ```
+
+#### Manually configure IP address and subnet for a node
+
+In the following scenarios, you may want to configure a specific IP and subnet:
+
+- Hosts with multiple external interfaces
+- Host interfaces with multiple IP addresses
+- Changes to cross subnet packet encapsulation
+- Changes to host IP address
+
+You can configure specific IP address and subnet for a node using environment variables or by updating the [Node resource]({{site.baseurl}}/{{page.version}}/reference/resources/node).
+
+##### Configure IP and subnet using environment variables
+
+To configure IP and subnet values using environment variables, use a `kubectl patch` command. For example:
+
+```
+kubectl patch ds -n kube-system calico-node --patch \
+    '{"spec": {"template": {"spec": {"containers": [{"name": "calico-node", "env": [{"name": "IP", "value": "10.0.2.10/24"}, {"name": "IP6", "value": "fd80:24e2:f998:72d6::/120"}]}]}}}}'
+```
+
+>**Note**: If the subnet is omitted, the defaults are: /32 (IPv4) and /128 (IPv6). We recommend that you include the subnet information for clarity when specifying IP addresses.
+{: .alert .alert-info}
+
+
+##### Configure IP and subnet using node resource
+
+You can also configure the IP address and subnet on a Node resource.
+
+>**Tip**: When configuring the IP address on a Node resource, you may want to disable IP address options or environment variables on the node. IP options on the container take precedence, and will overwrite the values you configure on the node resource.
+{: .alert .alert-info}
+
+Use **calicoctl** to query the current node configuration. For example:
```

Comment:
Could we show a `calicoctl patch` example here as well or instead of the "get / update" approach?

Reply:
Added, thanks